### PR TITLE
backports: for v1.2.0-beta.2

### PR DESCRIPTION
**Pull request by smira**
PRs backported:

- #6136 
- #6140 
- #6146 
- #6145 
- #6152 
- #6153 
- #6157 
- #6158 
- #6168
- #6167 
- #6170

### CODE REVIEW

**Review by Unix4ever — APPROVED** [review]

File: internal/integration/provision/upgrade.go [hunk 1]
```diff
@@ -73,20 +73,18 @@ type upgradeSpec struct {
...
-	previousRelease = "v1.0.6"
-	stableRelease   = "v1.1.0" // or soon-to-be-stable
+	previousRelease = "v1.1.2"
+	stableRelease   = "v1.2.0-beta.1" // or soon-to-be-stable
 	// The current version (the one being built on CI) is DefaultSettings.CurrentVersion.
 
-	previousK8sVersion = "1.23.6" // constants.DefaultKubernetesVersion in the previousRelease
-	stableK8sVersion   = "1.24.2" // constants.DefaultKubernetesVersion in the stableRelease
+	previousK8sVersion = "1.24.3"      // constants.DefaultKubernetesVersion in the previousRelease
+	stableK8sVersion   = "1.25.0-rc.1" // constants.DefaultKubernetesVersion in the stableRelease
```
Comment (frezbo) [inline internal/integration/provision/upgrade.go:81]: `-rc.1` or 1.25?

**Review by frezbo — APPROVED** [review]

File: internal/integration/provision/upgrade.go (hunk 1 above)
Comment (smira) [reply internal/integration/provision/upgrade.go:81]: stable is 1.2.0-beta.1, so whatever is there ;)

we'll test the upgrade from rc-1 to final 1.25.0 in the tests

**Pull request merged**